generator/src/FluentStageFactoryGenerator.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\CodeGenerator;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Builder\Type\StageInterface;
+use MongoDB\CodeGenerator\Definition\GeneratorDefinition;
+use Nette\PhpGenerator\ClassType;
+use Nette\PhpGenerator\Method;
+use Nette\PhpGenerator\Parameter;
+use Nette\PhpGenerator\PhpFile;
+use Nette\PhpGenerator\PhpNamespace;
+use Nette\PhpGenerator\TraitType;
+use ReflectionClass;
+
+use function array_key_last;
+use function array_map;
+use function assert;
+use function file_get_contents;
+use function implode;
+use function sprintf;
+
+/**
+ * Generates a fluent factory class for aggregation pipeline stages.
```

Comment:
```suggestion
 * Generates a fluent factory trait for aggregation pipeline stages.
```

Make a note to update other references of "class" to "trait" below. For instance, `createFluentFactoryClass()` should now be `createFluentFactoryTrait()`.

This very class might also be better named `FluentFactoryTraitGenerator()`.

Reply:
I want to keep "Stage" in the generator class name.

Reply:
Should the generated file be `FluentStageFactoryTrait.php` in that case? I'll defer to you, as I'm not sure if you have future plans for more traits.

Reply:
The file is in the Stage namespace.